bundler/lib/bundler/dsl.rb

```diff
@@ -249,6 +250,19 @@ def platforms(*platforms)
     end
     alias_method :platform, :platforms
 
+    # Name collision, working around it for a proof
+    def lock_platform(*platforms)
+      platforms.each do |p|
+        gem_platform = Gem::Platform.new(p)
+        # TODO: this is copied from lock
+        if gem_platform.to_s == "unknown"
+          Bundler.ui.warn "The platform `#{platform_string}` is unknown to RubyGems " \
```

Comment:
IMO this should raise here

Reply:
makes sense. I copied it from the other place we add_platform, in lock.rb.